pkg/cloud/openstack/clients/machineservice.go

```diff
@@ -201,16 +202,44 @@ func (is *InstanceService) InstanceCreate(name string, config *openstackconfigv1
 		}
 	}
 	userData := base64.StdEncoding.EncodeToString([]byte(cmd))
+
+	// if we are using Boot-from-volume, let's ignore image in base create option
+	image := config.Image
+	if config.RootVolume.Size > 0 {
+		image = ""
+	}
+
 	createOpts = servers.CreateOpts{
 		Name:             name,
-		ImageName:        config.Image,
+		ImageName:        image,
 		FlavorName:       config.Flavor,
 		AvailabilityZone: config.AvailabilityZone,
 		Networks:         nets,
 		UserData:         []byte(userData),
 		SecurityGroups:   config.SecurityGroups,
 		ServiceClient:    is.computeClient,
 	}
+
+	if config.RootVolume.Size > 0 {
```

Comment:
I would prefer either a check to ensure that the `SourceUUID` is in the struct rather than checking the size:
```
	if config.RootVolume.SourceUUID != "" {
```

We could also check the struct as well:
```
	if config.RootVolume != nil {
```

Reply:
ok, missed this :(, will update to config.RootVolume != nil {